### ISSUE TITLE
BUG: F2PY Compilation Failures under Meson with fixed-form/F77 file

### ISSUE THREAD

**Issue by ClarkMyWords** [numpy/numpy#26840]
### Describe the issue:

In general, the current release version of [meson](https://github.com/mesonbuild/meson), 1.4.1, is not capable of correctly handling .f77/fixed format Fortran files, in F2PY or otherwise. This is addressed in the newest release candidate, [1.5.0.rc2](https://github.com/mesonbuild/meson/releases/tag/1.5.0rc2), specifically implementing [this commit](https://github.com/mesonbuild/meson/commit/2058f63b4e2d67cbe835499a43d8415f3c1d940d).

However, even with this newest version, f2py still fails to compile .f77 files under meson. I have detailed this in an issue/support request on meson [here](https://github.com/mesonbuild/meson/issues/13381), but, briefly, even with the fix in 1.5.0.rc2, using the `-ffixed-form` flag, meson attempts to compile a .f90 file, which subsequently gets clobbered with C code, and causes the compiler to crash and the build to fail.

I was using a more complex .f77 file (an Abaqus UMAT), but I have been testing with a simple hello-world style .f77 program supplied by meson for testing, [simple.f77](https://github.com/user-attachments/files/16077437/simplef77.txt). Even this program does not compile under f2py/meson. This is the generated .pyf file I am using with this test case: 
[simple.pyf](https://github.com/user-attachments/files/16077450/simplepyf.txt)

In summary, while I have submitted a request for support from meson, what action can I take to use f2py with fixed-format Fortran? I tried, briefly, to install numpy==1.22.4 and use the deprecated distutils backend, but my system libraries are new enough that I was running into compilation errors when building the older numpy version. Ideally, I would prefer to continue to use numpy 2.*

### Reproduce the code example:

```python
# Using the above simple.f77 and simple.pyf:
export FFLAGS="-ffixed-form"
f2py -c simple.pyf simple.f77 --backend meson

# I receive the same error with f2py in a script, using subprocess.run to compile as encouraged in the documentation:
import os
import subprocess
os.environ["FFLAGS"] = "-ffixed-form"
subprocess.run([
    "f2py",
    "-c",
    "simple.pyf",
    "simple.f77",
    "--backend",
    "meson",
])
```

### Error message:

```shell
A copy of the error/output from meson:

Build started at 2024-07-02T23:11:44.858481
Main binary: /home/clark/Research/SinglePointSimulator/new_numpy_venv/bin/python
Build Options: 
Python system: Linux
The Meson build system
Version: 1.5.0.rc2
Source dir: /tmp/tmpm5k5u2ml
Build dir: /tmp/tmpm5k5u2ml/bbdir
Build type: native build
Project name: simple
Project version: 0.1
-----------
Detecting compiler via: `gfortran --version` -> 0
stdout:
GNU Fortran (GCC) 14.1.1 20240522
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-----------
pre-processor extraction using -cpp -x fortran failed, falling back w/o lang
Running command: -E -dM -
-----
-----------
Detecting linker via: `gfortran -Wl,--version -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form` -> 0
stdout:
GNU ld (GNU Binutils) 2.42.0
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
-----------
stderr:
collect2 version 14.1.1 20240522
/usr/bin/ld -plugin /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/lto-wrapper -plugin-opt=-fresolution=/tmp/cc13Z9oG.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/Scrt1.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crti.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtbeginS.o -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1 -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib -L/lib/../lib -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../.. --version -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtendS.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crtn.o
-----------
Sanity testing Fortran compiler: gfortran
Is cross compiler: False.
Sanity check compiler command line: gfortran sanitycheckf.f -o sanitycheckf.exe -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Fortran compiler for the host machine: gfortran (gcc 14.1.1 "GNU Fortran (GCC) 14.1.1 20240522")
Fortran linker for the host machine: gfortran ld.bfd 2.42.0
-----------
Detecting archiver via: `gcc-ar --version` -> 0
stdout:
GNU ar (GNU Binutils) 2.42.0
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) any later version.
This program has absolutely no warranty.
-----------
-----------
Detecting compiler via: `cc --version` -> 0
stdout:
cc (GCC) 14.1.1 20240522
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-----------
Running command: -cpp -x c -E -dM -
-----
-----------
Detecting linker via: `cc -Wl,--version` -> 0
stdout:
GNU ld (GNU Binutils) 2.42.0
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
-----------
stderr:
collect2 version 14.1.1 20240522
/usr/bin/ld -plugin /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/lto-wrapper -plugin-opt=-fresolution=/tmp/cc6vUJx1.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/Scrt1.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crti.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtbeginS.o -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1 -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib -L/lib/../lib -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../.. --version -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtendS.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crtn.o
-----------
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe -D_FILE_OFFSET_BITS=64
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
C compiler for the host machine: cc (gcc 14.1.1 "cc (GCC) 14.1.1 20240522")
C linker for the host machine: cc ld.bfd 2.42.0
-----------
Detecting compiler via: `gfortran --version` -> 0
stdout:
GNU Fortran (GCC) 14.1.1 20240522
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-----------
pre-processor extraction using -cpp -x fortran failed, falling back w/o lang
Running command: -E -dM -
-----
-----------
Detecting linker via: `gfortran -Wl,--version -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form` -> 0
stdout:
GNU ld (GNU Binutils) 2.42.0
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
-----------
stderr:
collect2 version 14.1.1 20240522
/usr/bin/ld -plugin /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/lto-wrapper -plugin-opt=-fresolution=/tmp/cc7UgrJS.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/Scrt1.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crti.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtbeginS.o -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1 -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib -L/lib/../lib -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../.. --version -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtendS.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crtn.o
-----------
Sanity testing Fortran compiler: gfortran
Is cross compiler: False.
Sanity check compiler command line: gfortran sanitycheckf.f -o sanitycheckf.exe -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command:  /tmp/tmpm5k5u2ml/bbdir/meson-private/sanitycheckf.exe
Fortran compiler for the build machine: gfortran (gcc 14.1.1 "GNU Fortran (GCC) 14.1.1 20240522")
Fortran linker for the build machine: gfortran ld.bfd 2.42.0
-----------
Detecting archiver via: `gcc-ar --version` -> 0
stdout:
GNU ar (GNU Binutils) 2.42.0
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) any later version.
This program has absolutely no warranty.
-----------
-----------
Detecting compiler via: `cc --version` -> 0
stdout:
cc (GCC) 14.1.1 20240522
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-----------
Running command: -cpp -x c -E -dM -
-----
-----------
Detecting linker via: `cc -Wl,--version` -> 0
stdout:
GNU ld (GNU Binutils) 2.42.0
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
-----------
stderr:
collect2 version 14.1.1 20240522
/usr/bin/ld -plugin /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/lto-wrapper -plugin-opt=-fresolution=/tmp/ccV32bzC.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/Scrt1.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crti.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtbeginS.o -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1 -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib -L/lib/../lib -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../.. --version -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/crtendS.o /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/crtn.o
-----------
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe -D_FILE_OFFSET_BITS=64
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command:  /tmp/tmpm5k5u2ml/bbdir/meson-private/sanitycheckc.exe
C compiler for the build machine: cc (gcc 14.1.1 "cc (GCC) 14.1.1 20240522")
C linker for the build machine: cc ld.bfd 2.42.0
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Program /home/clark/Research/SinglePointSimulator/new_numpy_venv/bin/python found: YES (/home/clark/Research/SinglePointSimulator/new_numpy_venv/bin/python)
Searching for 'python-3.12' via pkgconfig lookup in LIBPC
Pkg-config binary missing from cross or native file, or env var undefined.
Trying a default Pkg-config fallback at pkg-config
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Determining dependency 'python-3.12' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_LIBDIR]: /usr/lib/pkgconfig
env[PKG_CONFIG_PATH]: 
env[PKG_CONFIG]: /usr/bin/pkg-config
-----------
Called: `/usr/bin/pkg-config --modversion python-3.12` -> 0
stdout:
3.12
-----------
env[PKG_CONFIG_LIBDIR]: /usr/lib/pkgconfig
env[PKG_CONFIG_PATH]: 
env[PKG_CONFIG]: /usr/bin/pkg-config
-----------
Called: `/usr/bin/pkg-config --cflags python-3.12` -> 0
stdout:
-I/usr/include/python3.12
-----------
env[PKG_CONFIG_LIBDIR]: /usr/lib/pkgconfig
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
env[PKG_CONFIG]: /usr/bin/pkg-config
-----------
Called: `/usr/bin/pkg-config --libs python-3.12` -> 0
stdout:
-L/usr/lib
-----------
env[PKG_CONFIG_LIBDIR]: /usr/lib/pkgconfig
env[PKG_CONFIG_PATH]: 
env[PKG_CONFIG]: /usr/bin/pkg-config
-----------
Called: `/usr/bin/pkg-config --libs python-3.12` -> 0
Run-time dependency python found: YES 3.12
Running command: /home/clark/Research/SinglePointSimulator/new_numpy_venv/bin/python -c 'import os; os.chdir(".."); import numpy; print(numpy.get_include())'
--- stdout ---
/home/clark/Research/SinglePointSimulator/new_numpy_venv/lib/python3.12/site-packages/numpy/_core/include

--- stderr ---


Running command: /home/clark/Research/SinglePointSimulator/new_numpy_venv/bin/python -c 'import os; os.chdir(".."); import numpy.f2py; print(numpy.f2py.get_include())'
--- stdout ---
/home/clark/Research/SinglePointSimulator/new_numpy_venv/lib/python3.12/site-packages/numpy/f2py/src

--- stderr ---


Running compile:
Working directory:  /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpmzw9ln71
Code:
stop; end program
-----------
Command line: `gfortran /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpmzw9ln71/testfile.f90 -o /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpmzw9ln71/output.exe -D_FILE_OFFSET_BITS=64 -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form -O0 -lquadmath -Wl,--allow-shlib-undefined -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form` -> 1
stderr:
/tmp/tmpm5k5u2ml/bbdir/meson-private/tmpmzw9ln71/testfile.f90:1:2:

    1 | stop; end program
      |  1
Error: Non-numeric character in statement label at (1)
-----------
Running compile:
Working directory:  /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74
Code:

        #include<stddef.h>
        #include<stdio.h>
        int main(void) {
            printf("%ld\n", (long)(sizeof(void *)));
            return 0;
        }
-----------
Command line: `gfortran /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90 -o /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/output.exe -D_FILE_OFFSET_BITS=64 -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form -O0 -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form` -> 1
stderr:
/tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90:2:10:

    2 |         #include<stddef.h>
      |          1
Error: Invalid character in name at (1)
/tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90:3:10:

    3 |         #include<stdio.h>
      |          1
Error: Invalid character in name at (1)
/tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90:5:18:

    5 |             printf("%ld\n", (long)(sizeof(void *)));
      |                  1
Error: Syntax error in PRINT statement at (1)
/tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90:6:19:

    6 |             return 0;
      |                   1
Error: Alternate RETURN statement at (1) is only allowed within a SUBROUTINE
/tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90:7:10:

    7 |         }
      |          1
Error: Invalid character in name at (1)
-----------
Could not compile test file /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpcrh15p74/testfile.f90: 1

Running compile:
Working directory:  /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpfw4g9ues
Code:

-----------
Command line: `gfortran /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpfw4g9ues/testfile.f90 -o /tmp/tmpm5k5u2ml/bbdir/meson-private/tmpfw4g9ues/output.obj -D_FILE_OFFSET_BITS=64 -c -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form -O0 --print-search-dirs -fcray-pointer -cpp -fPIC -O3 -funroll-loops -ffixed-form` -> 0
stdout:
install: /usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/
programs: =/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../x86_64-pc-linux-gnu/bin/
libraries: =/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../x86_64-pc-linux-gnu/lib/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../x86_64-pc-linux-gnu/lib/../lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../x86_64-pc-linux-gnu/14.1.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../lib/:/lib/x86_64-pc-linux-gnu/14.1.1/:/lib/../lib/:/usr/lib/x86_64-pc-linux-gnu/14.1.1/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../../x86_64-pc-linux-gnu/lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/14.1.1/../../../:/lib/:/usr/lib/
-----------
Library quadmath found: NO
Build targets in project: 1

Found ninja-1.11.1.git.kitware.jobserver-1 at /home/clark/Research/SinglePointSimulator/new_numpy_venv/bin/ninja

ERROR: No specified compiler can handle file simple.f77
```

### Python and NumPy Versions:

2.0.0
3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]

### Runtime Environment:

[{'numpy_version': '2.0.0',
  'python': '3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]',
  'uname': uname_result(system='Linux', node='clarkpc', release='6.9.7-arch1-1', version='#1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000', machine='x86_64')},
 {'simd_extensions': {'baseline': ['SSE', 'SSE2', 'SSE3'],
                      'found': ['SSSE3',
                                'SSE41',
                                'POPCNT',
                                'SSE42',
                                'AVX',
                                'F16C',
                                'FMA3',
                                'AVX2'],
                      'not_found': ['AVX512F',
                                    'AVX512CD',
                                    'AVX512_KNL',
                                    'AVX512_KNM',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}},
 {'architecture': 'Zen',
  'filepath': '/home/clark/Temp/meson_bug/venv/lib/python3.12/site-packages/numpy.libs/libscipy_openblas64_-99b71e71.so',
  'internal_api': 'openblas',
  'num_threads': 16,
  'prefix': 'libscipy_openblas',
  'threading_layer': 'pthreads',
  'user_api': 'blas',
  'version': '0.3.27'}]

### Context for the issue:

I'm just wondering what the proper way(s) to compile .f77 Fortran/Fixed-Format Fortran Programs under f2py are? I don't know of any other tools that provides interoperability between Fortran and Python, and I am, for better or worse, stuck using fixed-format Fortran for some of my work. I understand that the issue is likely on Meson's end, hence why I submitted an issue to them before submitting this one, but I wanted to ensure this was reported for documentation purposes and to ask if there is a known/recommended workaround until Meson has a full release that correctly handles .f77.

Thank you for any and all help in advance!

**Comment by HaoZeke** [comment]
From an F2PY perspective, as long as the extension of the file is `.f`[1], `f2py` will / should correctly generate wrapper files, and these files will be exactly as compatible as the earlier `distutils` ones (since the wrapper generation is unchanged by the build backends). The next part (`meson` support) is probably an upstream issue.

[1] the reason this is important is because the wrapper generation needs to know if `fixed-form` code is being wrapped or not (`f77` mode)

**Comment by ClarkMyWords** [comment]
Thank you, @HaoZeke! I may not have been clear enough, but yes, my wrapper files are being written just fine, and the actual issue is actually upstream with `Meson`.

I've documented the issue with them at the issue I linked to in my initial post.

In the meantime, I've also developed [this gist](https://gist.github.com/ClarkMyWords/ca9b9a17151263eb405e1b76437c40cc) which implements an in-situ fix, patching the meson installation in-place to address this bug. Certainly not best-practice, but hopefully a useful workaround until a robust solution exists in Meson. (In specific, my use-case is with Abaqus UMAT files, which all have the same structure and functions, so this is by no means a complete fix, but it should at least work in many cases, and hopefully point others in the right direction).

With that said, that's all the support I needed. I just wanted to be sure that this issue was documented, that the f2py maintainers were aware, and that anyone who was like me looking for solutions could find this issue. Otherwise, I think this issue is fine to be closed, unless there's anything I'm missing.

Thanks again!
